org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/server/wizard/ServerConnectionWizard.java

```diff
@@ -190,15 +192,28 @@ public boolean performFinish() {
       return false;
     }
     IServer server;
+
     if (model.isEdit()) {
       editedServer.updateConfig(model.getServerUrl(), model.getOrganization(), model.getUsername(), model.getPassword(), model.getNotificationsEnabled());
       server = editedServer;
 
-      if (model.getNotificationsSupported() && model.getNotificationsEnabled()) {
-        server.getBoundProjects().forEach(SonarLintUiPlugin::subscribeToNotifications);
+      Job job = new ServerUpdateJob(server);
+
+      List<ISonarLintProject> boundProjects = server.getBoundProjects();
+      if (model.getNotificationsSupported() && model.getNotificationsEnabled() && !boundProjects.isEmpty()) {
+        Job subscribeToNotificationsJob = new Job("Subscribe to notifications") {
+          @Override
+          protected IStatus run(IProgressMonitor monitor) {
+            boundProjects.forEach(SonarLintUiPlugin::subscribeToNotifications);
+            return Status.OK_STATUS;
+          }
+        };
+        JobUtils.scheduleAfter(job, () -> subscribeToNotificationsJob.schedule());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1612)